stake-pool/program/src/processor.rs

```diff
@@ -2005,9 +2007,7 @@ impl Processor {
             program_id,
             stake_pool_info.key,
         )?;
-        if let Some(sol_deposit_authority) = sol_deposit_authority_info {
-            stake_pool.check_sol_deposit_authority(sol_deposit_authority)?;
-        }
+        stake_pool.check_sol_deposit_authority(sol_deposit_authority_info)?;
```

Comment:
Hmm, I'm not sure this is the behaviour we want, shouldn't the sol deposit authority be optional?

Reply:
Oh wait, I missed that the ? operator only triggers the error if there is indeed Some(deposit_sol_auth)